### ISSUE TITLE
-InputObject Parameter Support for New-AzRoleDefinition and Set-AzRoleDefinition cmdlets

### ISSUE THREAD

**Issue by Duffney** [Azure/azure-powershell#14522]
## Description of the new feature

Add the `-InputObject` parameter to the appropriate AzRoleDefintion cmdlets.

Reason: Currently you can either input a JSON file or a RoleDefinition object. Manipulating a RoleDefinition object is complicated, and the input file requires you store a local copy of the definition in JSON on disk. Adding the `InputObject` parameter would make it easier for operates and developers to manipulate the object without having to go through the additional steps of storing it on disk and then removing it for cleanup purposes.

**TL: DR Provides a cleaner way to manipulate and interact with role definitions.**

## Proposed implementation details (optional)

Add a parameter to the cmdlet binding that converts the PowerShell object passed to it into JSON. An example of that operation can be found [here](https://github.com/Duffney/InvokeDSC/blob/master/InvokeDSC/Public/ConvertTo-DSC.ps1), if interested.

**Comment by EwertonJordao** [comment]
In my case I try to include a valid json to create a role, but I recieve that exception: 
```console
ErrorRecord    : File {
                   "Name": "OpenCost Role",
                   "IsCustom": true,
                   "Description": "Rate read role",
                   "Actions": [
                     "Microsoft.Compute/virtualMachines/vmSizes/read",
                     "Microsoft.Resources/subscriptions/locations/read",
                     "Microsoft.Resources/providers/read",
                     "Microsoft.ContainerService/containerServices/read",
                     "Microsoft.Commerce/RateCard/read"
                   ],
                   "AssignableScopes": [
                     "/subscriptions/0000000-0000-0000-0000-000000000000"
                   ]
                 } does not exist
Message        : File {
                   "Name": "OpenCost Role",
                   "IsCustom": true,
                   "Description": "Rate read role",
                   "Actions": [
                     "Microsoft.Compute/virtualMachines/vmSizes/read",
                     "Microsoft.Resources/subscriptions/locations/read",
                     "Microsoft.Resources/providers/read",
                     "Microsoft.ContainerService/containerServices/read",
                     "Microsoft.Commerce/RateCard/read"
                   ],
                   "AssignableScopes": [
                     "/subscriptions/0000000-0000-0000-0000-000000000000"
                    
                   ]
                 } does not exist
ParamName      : 
TargetSite     : Void ExecuteCmdlet()
Data           : {}
InnerException : 
HelpLink       : 
Source         : Microsoft.Azure.PowerShell.Cmdlets.Resources
HResult        : -2147024809
StackTrace     :    at Microsoft.Azure.Commands.Resources.NewAzureRoleDefinitionCommand.ExecuteCmdlet()
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.CmdletExtensions.<>c__3`1.<ExecuteSynchronouslyOrAsJ
                 ob>b__3_0(T c)
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.CmdletExtensions.ExecuteSynchronouslyOrAsJob[T](T cm
                 dlet, Action`1 executor)
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.CmdletExtensions.ExecuteSynchronouslyOrAsJob[T](T cm
                 dlet)
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()
````
This is a json valid but the implementation need to suport new param for this implementation more clear than now:

https://docs.microsoft.com/en-us/powershell/module/az.resources/New-azRoleDefinition?view=azps-8.3.0#example-1-create-using-psroledefinitionobject 

sample implementation expected: 

```powershell
$subscriptioID=$(Get-AzContext).Subscription.Id
$roleDefinitionContent=[PSCustomObject]@{
    Name = "OpenCost Role"
    IsCustom = $true
    Description = "Rate read role"
    Actions=@(
        "Microsoft.Compute/virtualMachines/vmSizes/read",
        "Microsoft.Resources/subscriptions/locations/read",
        "Microsoft.Resources/providers/read",
        "Microsoft.ContainerService/containerServices/read",
        "Microsoft.Commerce/RateCard/read"
    )
    AssignableScopes=@(
        "/subscriptions/$subscriptioID"
    )
} | ConvertTo-Json 
New-AzRoleDefinition -RoleInput $roleDefinitionContent
```

**Comment by MaddyMicrosoft** [comment]
Hi @duffney, adding -InputObject to New-AzRoleDefinition isn't on the roadmap right now. The supported paths remain -Role (typed PSRoleDefinition) and -InputFile (JSON), and Set-AzRoleDefinition accepts pipeline via -Role for the update case.

If you want to avoid the disk round-trip, one thing you could try is deserializing a JSON string directly into a PSRoleDefinition via Newtonsoft.Json (not officially supported, but it's a common pattern for PowerShell-hosted deserialization).

Given the limited demand signal and the existing options, this is a nice-to-have without enough return to prioritize the new parameter surface. Closing on that basis. Happy to revisit if it becomes blocking for a specific workflow.
